From 71cdfd9899c29d5692db12ef57ba18378059b726 Mon Sep 17 00:00:00 2001
From: JasonYANG170 <qw200477@163.com>
Date: Sun, 19 Jul 2026 18:11:44 +0800
Subject: [PATCH] arm64: dts: rockchip: PineTab2 v0.1 disable display, WiFi,
 camera, touch, amplifier

Remove the following components from PineTab2 v0.1 device tree:
- Display: LCD, DSI, HDMI, backlight, VOP
- WiFi: SDIO interface (sdmmc1)
- Camera: OV5648 sensor, CSI DPHY
- Touchscreen: Goodix GT911
- Amplifier: speaker amplifier, I2S audio

This is useful for headless/embedded use cases where these
peripherals are not needed.
---
 .../dts/rockchip/rk3566-pinetab2-v0.1.dts     | 90 +++++++++++++++++--
 1 file changed, 81 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-pinetab2-v0.1.dts b/arch/arm64/boot/dts/rockchip/rk3566-pinetab2-v0.1.dts
index 8b110186a3eb..d7ba17e3b2a0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-pinetab2-v0.1.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-pinetab2-v0.1.dts
@@ -7,20 +7,92 @@
 / {
 	model = "Pine64 PineTab2 v0.1";
 	compatible = "pine64,pinetab2-v0.1", "pine64,pinetab2", "rockchip,rk3566";
+
+	/delete-node/ hdmi-con;
+	/delete-node/ led-flash;
+	/delete-node/ led_flash_reg;
+	/delete-node/ rk817-sound;
+	/delete-node/ speaker_amp;
+	/delete-node/ vdd1v2_dvp;
+};
+
+/* Disable display: LCD, DSI, HDMI */
+&backlight {
+	status = "disabled";
+};
+
+&dsi0 {
+	status = "disabled";
+};
+
+&dsi0_in {
+	status = "disabled";
+};
+
+&dsi0_out {
+	status = "disabled";
+};
+
+&dsi_dphy0 {
+	status = "disabled";
+};
+
+&hdmi {
+	status = "disabled";
+};
+
+&hdmi_in {
+	status = "disabled";
+};
+
+&hdmi_out {
+	status = "disabled";
 };
 
-&lcd {
-	reset-gpios = <&gpio0 RK_PA6 GPIO_ACTIVE_LOW>;
+&hdmi_sound {
+	status = "disabled";
 };
 
-&pinctrl {
-	lcd0 {
-		lcd0_rst_l: lcd0-rst-l {
-			rockchip,pins = <0 RK_PA6 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
+&vop {
+	status = "disabled";
 };
 
+&vop_mmu {
+	status = "disabled";
+};
+
+&vp0 {
+	status = "disabled";
+};
+
+&vp1 {
+	status = "disabled";
+};
+
+/* Disable camera */
+&csi_dphy {
+	status = "disabled";
+};
+
+&i2c2 {
+	status = "disabled";
+};
+
+/* Disable touchscreen */
+&i2c1 {
+	status = "disabled";
+};
+
+/* Disable WiFi (SDIO) */
 &sdmmc1 {
-	vmmc-supply = <&vcc3v3_sys>;
+	status = "disabled";
+};
+
+/* Disable audio/speaker amplifier */
+&i2s0_8ch {
+	status = "disabled";
+};
+
+&i2s1_8ch {
+	status = "disabled";
 };
-- 
2.51.0

